tests/rptest/tests/raft_availability_test.py

```diff
@@ -267,60 +283,36 @@ def test_one_node_down(self):
         self.logger.info(f"Leadership moved to {new_leader}")
 
         self._expect_available()
-
-        # Metric checks are relaxed to tolerate a particular bug where redpanda sometimes
-        # goes through multiple elections to select a new leader, even though it should
-        # be able to do it in 1.
-        # Ref: https://github.com/vectorizedio/redpanda/issues/2444
-        if False:
-            observer_metrics.expect([
-                # 'leadership changes' increments by 1 when observer sees an append_entries
-                # from the new leader
-                ("vectorized_raft_leadership_changes_total",
-                 lambda a, b: b == a + 1),
-                ("vectorized_raft_leader_for", lambda a, b: int(b) == 0),
-                ("vectorized_raft_received_vote_requests_total",
-                 lambda a, b: b == a + 2),
-            ])
-
-            new_leader_metrics.expect([
-                # 'leadership changes' includes going to candidate, then going to leader, so
-                # increments by 2 (vote_stm::vote calls trigger_leadership_notification when
-                # we start an election, before the leadership has really changed)
-                ("vectorized_raft_leadership_changes_total",
-                 lambda a, b: b == a + 2),
-                ("vectorized_raft_leader_for", lambda a, b: int(b) == 1),
-
-                # The new leader should see heartbeat errors sending to the now-offline
-                # original leader
-                ("vectorized_raft_heartbeat_requests_errors_total",
-                 lambda a, b: b > a),
-
-                # This node initiated the vote, so it should not have received any votes
-                ("vectorized_raft_received_vote_requests_total",
-                 lambda a, b: b == a),
-            ])
-        else:
-            # Relaxed checks that don't care how many elections happened, just that
-            # >1 happened.  Tolerant of the cluster going through spurious elections even
-            # though it should elect a new leader in a single pass.
-            observer_metrics.expect([
-                ("vectorized_raft_leadership_changes_total",
-                 lambda a, b: b > a),
-                ("vectorized_raft_leader_for", lambda a, b: int(b) == 0),
-                ("vectorized_raft_received_vote_requests_total",
-                 lambda a, b: b > a),
-            ])
-
-            new_leader_metrics.expect([
-                ("vectorized_raft_leadership_changes_total",
-                 lambda a, b: b > a),
-                ("vectorized_raft_leader_for", lambda a, b: int(b) == 1),
-                ("vectorized_raft_heartbeat_requests_errors_total",
-                 lambda a, b: b > a),
-                ("vectorized_raft_received_vote_requests_total",
-                 lambda a, b: b == a),
-            ])
+        time.sleep(0.1)
```

Comment:
This sleep is to reduce chance of vectorized_raft_heartbeat_requests_errors_total not having incremented yet, I'm guessing?  Please can you add a comment to explain (otherwise sleeps in tests look like they're just there for luck :-) )

Reply:
The test relies on luck, TBH. :laughing: Happy to add another comment though.